src/pagination/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import theme from '@dojo/framework/core/middleware/theme';
+import { WNode, RenderResult } from '@dojo/framework/core/interfaces';
+import { createResource, DataTemplate } from '@dojo/framework/core/resource';
+
+import Icon from '../icon';
+import Select, { defaultTransform } from '../select';
+
+import bundle from './Pagination.nls';
+import * as css from '../theme/default/pagination.m.css';
+
+function createMemoryTemplate<S = void>(): DataTemplate<S> {
+	return {
+		read: (_, put, get) => {
+			let data: any[] = get();
+			put(0, data);
+			return { data, total: data.length };
+		}
+	};
+}
+
+const memoryTemplate = createMemoryTemplate();
+
+export interface PaginationProperties {
+	/** The initial page number */
+	initialPage?: number;
+
+	/** The initial page size */
+	initialPageSize?: number;
+
+	/** Callback fired when the page number is changed */
+	onPageChange(page: number): void;
+
+	/** Callback fired when the page size is changed */
+	onPageSizeChange?(pageSize: number): void;
+
+	/** Page size options to display in the page size selector */
+	pageSizes?: number[];
+
+	/** Custom renderer for page size selector item */
+	pageSizeRenderer?(pageSize: number): RenderResult;
```

Comment:
I'm not sure how valid this is as a use case. I think that 99.9% of the time we'd just want to see the page numbers in the drop down, not even `30 / page`, just `10, 20, 30` etc...